python/dgl/partition.py

```diff
@@ -497,5 +514,35 @@ def map_to_global(self, idxs, part_id):
         return F.zerocopy_from_dgl_ndarray(_CAPI_DGLNDArrayPartitionMapToGlobal(
             self._partition, F.zerocopy_to_dgl_ndarray(idxs), part_id))
 
+    def mode(self):
+        """ Get the type of partition.
+        """
+        return self._mode
+
+def create_edge_partition_from_nodes(partition, graph):
+    """ Create a new partition of the edges in the graph from the partition of
+        the vertices. Edges may not be assigned to the same partition as their
+        source or destination vertex.
+    """
+    if partition.mode() == 'remainder':
+        # This mode has no locality, so we don't bother to assign vertices to
+        # the same partition
+        return NDArrayPartition(array_size=graph.number_of_edges(),
+                                num_parts=partition.num_parts(),
+                                mode=partition._mode)
+    else:
+        assert partition.mode() == 'range'
+        # This mode makes use of locality, so we want to preserve the same
+        # locality for edges. To do this, we find the number of edges
+        # per-partition and use the prefix-sum of this for the range
+        rng = [0]
+        for part in range(graph.num_parts()):
+            idx = partition.get_local_indices(part, ctx=F.cpu())
+            rng.append(rng[-1] + \
+                F.as_scalar(F.sum(graph.out_degrees(u=idx), dim=0)))
+        return NDArrayPartition(array_size=graph.number_of_edges(),
+                                num_parts=partition.num_parts(),
```

Comment:
What if each node has only one partition but has more than one GPUs?

Reply:
The number of GPUs must be equal to the number of partitions. Or are you asking if the number of nodes in the graph is less than the number of partitions?